### ISSUE TITLE
Native notifications don’t work anymore

### ISSUE THREAD

**Issue by jurf** [flathub/org.telegram.desktop#55]
Not a deal-breaker for me, but a bit of a nuisance.

![obrazok](https://user-images.githubusercontent.com/7472425/51406777-5a72f280-1b5a-11e9-9a02-d6afd992d0df.png)

**Comment by eternal-sorrow** [comment]
@grulja for me native notifications still do not work (no such option in settings, just like OP had). Updating to latest version did not help.

**Comment by eternal-sorrow** [comment]
Here is how it looks now:
![screenshot](https://cdn.pbrd.co/images/HZyaSJy.png)

**Comment by grulja** [comment]
It works for me. Maybe you don't have any org.kde.StatusNotifierWatcher implementation? In Gnome you need some extension.

**Comment by eternal-sorrow** [comment]
> org.kde.StatusNotifierWatcher

This one is for tray, isn't it? What it has to do with notifications? Telegram does not use libnotify for this?

**Comment by grulja** [comment]
Right, I from some reason assumed this is about tray icon. Telegram appears to be using libnotify.

**Comment by eternal-sorrow** [comment]
So? How do I solve the native notifications not working?

**Comment by grulja** [comment]
I unfortunately have no idea, do you have latest telegram? There is now 1.5.11 available in Flathub.

**Comment by eternal-sorrow** [comment]
Yes. The first thing I did when I discovered this problem is update telegram flatpak to the latest version. Now I've got 1.5.11.

**Comment by jurf** [comment]
Have you enabled it in the settings?

**Comment by eternal-sorrow** [comment]
As I said already, there is no such option in settings.

**Comment by eternal-sorrow** [comment]
So? I updated to 1.6.2 and it still does not work.

**Comment by eternal-sorrow** [comment]
1.6.7 no option for native notifications

**Comment by jurf** [comment]
What system are you running? What is your setup?

**Comment by eternal-sorrow** [comment]
Gentoo, [sway](https://github.com/swaywm/sway) wayland compositor, [mako](https://github.com/emersion/mako) notification daemon.

**Comment by eternal-sorrow** [comment]
1.7 still no desktop notifications.

I wonder, why there is no notify-send and libnotify binaries in telegram sandbox?

**Comment by grulja** [comment]
I have native notifications in 1.7. You need to make sure your desktop does implement org.freedesktop.Notifications support.

**Comment by eternal-sorrow** [comment]
> You need to make sure your desktop does implement org.freedesktop.Notifications support.
Notifications work in native (non-flatpak) desktop applications, so I guess I have this support.

**Comment by eternal-sorrow** [comment]
So? Is there any way to further debug why native notifications do not work?

**Comment by eternal-sorrow** [comment]
Okay, I figured it myself. I ran telegram with debug logs enabled and that's what I got in the log file, among other lines:
```
[2019.05.31 01:34:18] Could not load 'ayatana-appindicator3' with version 1 :(
[2019.05.31 01:34:18] Could not load 'ayatana-appindicator' with version 1 :(
[2019.05.31 01:34:18] Could not load 'appindicator' with version 1 :(
[2019.05.31 01:34:18] Could not load gtk-x11-2.0!
```

Looking at the tdesktop source code I can see, that if appindicator and gtk libraries are not loaded, then it doesn't even try to load libnotify. I don't understand very well how the flatpak works, so can anyone explain me, why this doesn't work?

**Comment by eternal-sorrow** [comment]
Okay, tdesktop's logging system is a bit odd, so I got confused but I finally figured it out. I was able to read more verbose debug logs, so I narrowed the error down to the line:
```
Failed to gtk_init_check(0, 0)!
```

**Comment by eternal-sorrow** [comment]
Okay, I guess I need to report this as separate issue.
